nix/mobile/default.nix

```diff
@@ -1,38 +1,42 @@
-{ config, stdenv, pkgs, callPackage, mkShell, fetchurl, fetchFromGitHub, target-os,
-  mkFilter, localMavenRepoBuilder, maven, status-go, composeXcodeWrapper, nodejs, yarn }:
+{ config, lib, stdenvNoCC, callPackage,status-go, 
+  localMavenRepoBuilder, composeXcodeWrapper, mkShell, mergeSh }:
 
 let
-  inherit (stdenv.lib) catAttrs concatStrings optional unique;
+  inherit (lib) catAttrs concatStrings optional unique;
 
-  platform = callPackage ../platform.nix { inherit target-os; };
   xcodewrapperArgs = {
     version = "11.2.1";
   };
   xcodeWrapper = composeXcodeWrapper xcodewrapperArgs;
-  androidPlatform = callPackage ./android { inherit config target-os mkShell mkFilter nodejs maven localMavenRepoBuilder projectNodePackage jsbundle; status-go = status-go.android; };
-  iosPlatform = callPackage ./ios { inherit config mkFilter mkShell xcodeWrapper projectNodePackage fastlane; status-go = status-go.ios; };
-  fastlane = callPackage ./fastlane { inherit stdenv target-os mkShell; };
+  fastlane = callPackage ./fastlane { };
+  androidPlatform = callPackage ./android {
+    inherit localMavenRepoBuilder projectNodePackage;
+    status-go = status-go.android;
+  };
+  iosPlatform = callPackage ./ios {
+    inherit xcodeWrapper projectNodePackage fastlane;
+    status-go = status-go.ios;
+  };
   selectedSources = [
-      fastlane
-      status-go.android
-      status-go.ios
-    ] ++
-    optional platform.targetAndroid androidPlatform ++
-    optional platform.targetIOS iosPlatform;
+    fastlane
+    status-go.android
+    status-go.ios
+    androidPlatform 
```

Comment:
If I start a `make shell TARGET=ios` on a clean system, will that download Android stuff (it wasn't previously)?

Reply:
Hmmm, it might, can you test that?

Reply:
I've tested it and only see `/nix/store/0nvv75l3zpw421l4v2yzwdy47cnngmp7-jsc_android___jsc_android_245459.0.0.tgz`. It's not too bad. There are 2 downsides with not knowing what we're targeting: unnecessary bandwidth(+ time) and unnecessary dependencies (e.g. if an Android dep becomes broken on macOS we break the iOS env too). We might just decide we prefer living with that than with the complexity of `target-os`.

Reply:
At least I don't see the NDK being pulled, so I guess we're fine here. That would be a huge dep to pull unnecessarily.

Reply:
I will have to think about this one tomorrow.

Reply:
I think it looks correct:
```
jenkins@macos-01.ms-eu-dublin.ci.misc:~/repos/status-react % make shell TARGET=ios
Configuring default Nix shell for target 'ios'...
Checking for modifications in node_modules...
No modifications detected.
Checking for modifications in ios/Pods...
No modifications detected.

[nix-shell:~/repos/status-react]$ echo $buildInputs | tr ' ' '\n'
/nix/store/w6l0vw6yzss9lc8cvxzyb8z2ajhhwi9g-xcode-wrapper-11.2.1
/nix/store/d6dg4pl5laprgchzhzhx58z0bj91pbbj-watchman-4.9.0
/nix/store/1lnv10pvii0s8snmfw2snb4dv4rnj5xr-bundler-1.17.3
/nix/store/x6hwh3h408k4c28sir7askm3fqpl4hb0-procps-1003.1-2008
/nix/store/bjpxsdfjqd51mglpsl9pz1gdw7a4fpz5-flock-0.2.3
/nix/store/x5pzk8c516g6k46gxq3v8jxipgg6vvlf-status-go-0.36.1-fd49b01-ios
/nix/store/vp6r9vqbf5pig2avhacwz7hiw4xwcd39-ruby-2.6.5
/nix/store/1lnv10pvii0s8snmfw2snb4dv4rnj5xr-bundler-1.17.3
/nix/store/zf6hzp1xn20fmf23s6k1vc6ig6s65xqa-fastlane-2.133.0
/nix/store/8627xzqkrf5cp23a8g4q9s87aa8h2idb-curl-7.67.0-bin
/nix/store/iyv7b8781cqh0r4v5fxzgnyfvqbjmdcw-cocoapods-1.8.4
/nix/store/vp6r9vqbf5pig2avhacwz7hiw4xwcd39-ruby-2.6.5
/nix/store/1lnv10pvii0s8snmfw2snb4dv4rnj5xr-bundler-1.17.3
/nix/store/zf6hzp1xn20fmf23s6k1vc6ig6s65xqa-fastlane-2.133.0
/nix/store/09jcdiv7fqj2jx4r4vpxbr5nrr8rfzaa-curl-7.67.0-dev
/nix/store/iyv7b8781cqh0r4v5fxzgnyfvqbjmdcw-cocoapods-1.8.4
/nix/store/iyv7b8781cqh0r4v5fxzgnyfvqbjmdcw-cocoapods-1.8.4
/nix/store/5pp98fn44axh3gdl68naba8ryak0bh3p-bash-4.4-p23-dev
/nix/store/09jcdiv7fqj2jx4r4vpxbr5nrr8rfzaa-curl-7.67.0-dev
/nix/store/9a5a676m264kbbd0hpwgl3bkxs2b56mx-wget-1.20.3
/nix/store/0lzhngrf5dsa4xrrz6ssz7qganb6cm02-file-5.37
/nix/store/15wng30fs9nbzfd2xcsdypi0qip8m1s8-unzip-6.0
/nix/store/bjpxsdfjqd51mglpsl9pz1gdw7a4fpz5-flock-0.2.3
/nix/store/90g2i8jcalvngb7hj1a1vk57vjbmdzb4-git-2.24.0
/nix/store/6ndn9v9id1j2gdbndi7x0bh53s0h7l7m-gnumake-4.2.1
/nix/store/2865ahxs2sj43vlsmzw7zcndm1fcxjbd-jq-1.6-dev
/nix/store/25fjxm795r5fn4pby15lhibh3gb4mf6b-ncurses-6.1-20190112-dev
/nix/store/0dj2sd7c6h3kn8ik9i3kcm0x8i4zarjb-clojure-1.10.0.411
/nix/store/1k2xnwi4nfn0q7j518aq80zk1vf8zys6-leiningen-2.9.1
/nix/store/y2vd5zk2pn3j36kkjdfsllxw10z2rccb-apache-maven-3.6.2
/nix/store/d6dg4pl5laprgchzhzhx58z0bj91pbbj-watchman-4.9.0
/nix/store/a7az4bx32yy968nw8c3ppsqdv0kv3mfd-yarn-1.19.1
/nix/store/dfjphxv0jbzi7fg9h6l5q6ri15hz2skz-nodejs-10.17.0
/nix/store/ylvwym6krs4wz6lxdpr458bpjpr6s08d-python-2.7.17
/nix/store/iyv7b8781cqh0r4v5fxzgnyfvqbjmdcw-cocoapods-1.8.4
/nix/store/60hjqrb8dn47jrffgcdzn85bz3bzbp24-clang-wrapper-7.1.0
```